src/SDAO.t.sol

```diff
@@ -0,0 +1,509 @@
+        deadline = bound(deadline, block.timestamp, block.timestamp + type(uint80).max);
+        privateKey = uint248(bound(privateKey, 1, type(uint248).max));
+
+        address owner = vm.addr(privateKey);
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+            privateKey,
+            keccak256(
+                abi.encodePacked(
+                    "\x19\x01",
+                    token.DOMAIN_SEPARATOR(),
+                    keccak256(abi.encode(PERMIT_TYPEHASH, owner, to, amount, 0, deadline))
+                )
+            )
+        );
+
+        token.permit(owner, to, amount, deadline, v, r, s);
+
+        assertEq(token.allowance(owner, to), amount);
+        assertEq(token.nonces(owner), 1);
+    }
+
+    function testRevertBurnInsufficientBalanceFuzz(address to, uint256 mintAmount, uint256 burnAmount) public {
+        vm.assume(to != address(0) && to != address(token));
+        mintAmount = bound(mintAmount, 0, type(uint256).max - 1);
+        burnAmount = bound(burnAmount, mintAmount + 1, type(uint256).max);
+
+        token.mint(to, mintAmount);
+
+        vm.expectRevert("SDAO/insufficient-balance");
+        token.burn(to, burnAmount);
+    }
+
+    function testRevertTransferInsufficientBalanceFuzz(address to, uint256 mintAmount, uint256 sendAmount) public {
+        vm.assume(to != address(0) && to != address(token));
+        mintAmount = bound(mintAmount, 0, type(uint256).max - 1);
+        sendAmount = bound(sendAmount, mintAmount + 1, type(uint256).max);
+
+        token.mint(address(this), mintAmount);
+
+        vm.expectRevert("SDAO/insufficient-balance");
+        token.transfer(to, sendAmount);
+    }
+
+    function testRevertTransferFromInsufficientAllowanceFuzz(address to, uint256 allowance, uint256 amount) public {
+        vm.assume(to != address(0) && to != address(token));
+        allowance = bound(allowance, 0, type(uint256).max - 1);
+        amount = bound(amount, allowance + 1, type(uint256).max);
+
+        address from = address(0xABCD);
+        token.mint(from, amount);
+        vm.prank(from);
+        token.approve(address(this), allowance);
+
+        vm.expectRevert("SDAO/insufficient-allowance");
+        token.transferFrom(from, to, amount);
+    }
+
+    function testRevertTransferFromInsufficientBalanceFuzz(address to, uint256 mintAmount, uint256 sendAmount) public {
+        vm.assume(to != address(0) && to != address(token));
+        mintAmount = bound(mintAmount, 0, type(uint256).max - 1);
+        sendAmount = bound(sendAmount, mintAmount + 1, type(uint256).max);
+
+        address from = address(0xABCD);
+        token.mint(from, mintAmount);
+        vm.prank(from);
+        token.approve(address(this), sendAmount);
+
+        vm.expectRevert("SDAO/insufficient-balance");
+        token.transferFrom(from, to, sendAmount);
+    }
+
+    function testRevertPermitBadNonceFuzz(
+        uint248 privateKey,
```

Comment:
Again I'm wondering why this is being reduced to `uint248`?

Reply:
Answered above.